src/traces/pie/plot.js

```diff
@@ -95,68 +104,78 @@ module.exports = function plot(gd, cdpie) {
-                    if(hoverinfo.indexOf('value') !== -1) thisText.push(helpers.formatPieValue(pt.v, separators));
-                    if(hoverinfo.indexOf('percent') !== -1) thisText.push(helpers.formatPiePercent(pt.v / cd0.vTotal, separators));
-
-                    var hoverLabel = trace.hoverlabel;
-                    var hoverFont = hoverLabel.font;
-
-                    Fx.loneHover({
-                        x0: hoverCenterX - rInscribed * cd0.r,
-                        x1: hoverCenterX + rInscribed * cd0.r,
-                        y: hoverCenterY,
-                        text: thisText.join('<br>'),
-                        name: hoverinfo.indexOf('name') !== -1 ? trace2.name : undefined,
-                        idealAlign: pt.pxmid[0] < 0 ? 'left' : 'right',
-                        color: helpers.castOption(hoverLabel.bgcolor, pt.pts) || pt.color,
-                        borderColor: helpers.castOption(hoverLabel.bordercolor, pt.pts),
-                        fontFamily: helpers.castOption(hoverFont.family, pt.pts),
-                        fontSize: helpers.castOption(hoverFont.size, pt.pts),
-                        fontColor: helpers.castOption(hoverFont.color, pt.pts)
-                    }, {
-                        container: fullLayout2._hoverlayer.node(),
-                        outerContainer: fullLayout2._paper.node(),
-                        gd: gd
-                    });
-
-                    Fx.hover(gd, evt, 'pie');
 
-                    hasHoverData = true;
+                    gd.emit('plotly_hover', {
+                        points: [eventData(pt, trace2)],
+                        event: d3.event
+                    });
+                    hasHoverEvent = true;
                 }
 
                 function handleMouseOut(evt) {
-                    evt.originalEvent = d3.event;
-                    gd.emit('plotly_unhover', {
-                        event: d3.event,
-                        points: [evt]
-                    });
+                    var fullLayout2 = gd._fullLayout;
+                    var trace2 = gd._fullData[trace.index];
+
+                    if(hasHoverEvent) {
+                        evt.originalEvent = d3.event;
+                        gd.emit('plotly_unhover', {
+                            points: [eventData(pt, trace2)],
+                            event: d3.event
+                        });
+                        hasHoverEvent = false;
+                    }
 
-                    if(hasHoverData) {
-                        Fx.loneUnhover(fullLayout._hoverlayer.node());
-                        hasHoverData = false;
+                    if(hasHoverLabel) {
+                        Fx.loneUnhover(fullLayout2._hoverlayer.node());
+                        hasHoverLabel = false;
                     }
                 }
 
                 function handleClick() {
-                    gd._hoverdata = [pt];
-                    gd._hoverdata.trace = cd0.trace;
```

Comment:
We probably didn't mean to attach `trace` as a property of the *whole array* 🙈

Reply:
Wow. Brutal. Good catch :baseball: